chat-server/src/services/embeddings.ts

```diff
@@ -1,42 +1,95 @@
-import { Configuration, OpenAIApi } from "openai";
+import { OpenAIClient } from "../integrations/openai";
+import { logger } from "./logger";
+import dotenv from "dotenv";
+import { stripIndent } from "common-tags";
+dotenv.config();
+
+interface EmbeddingParams {
+  text: string;
+  userIp: string;
+}
+
+interface EmbeddingResponse {
+  status: number;
+  errorData?: any;
+  embeddings?: number[];
+}
 
 class EmbeddingService {
   private embeddingProvider: EmbeddingProvider;
   constructor(embeddingProvider: EmbeddingProvider) {
     this.embeddingProvider = embeddingProvider;
   }
 
-  async createEmbedding(text: string) {
-    return await this.embeddingProvider.createEmbedding(text);
+  async createEmbedding({ text, userIp }: EmbeddingParams) {
+    return await this.embeddingProvider.createEmbedding({ text, userIp });
   }
 }
 
 // Abstract interface for embedding provider to make it easier to swap out
 // different providers in the future.
 abstract class EmbeddingProvider {
-  abstract createEmbedding(text: string): Promise<number[]>;
+  abstract createEmbedding({
+    text,
+    userIp,
+  }: EmbeddingParams): Promise<EmbeddingResponse>;
 }
 
-class OpenAIEmbeddingProvider extends EmbeddingProvider {
-  private openaiClient: OpenAIApi;
+export class OpenAIEmbeddingProvider extends EmbeddingProvider {
+  private openaiClient: OpenAIClient;
 
-  constructor(endpoint: string, apiKey: string) {
+  constructor(openaiClient: OpenAIClient) {
     super();
-    const configuration = new Configuration({
-      basePath: endpoint, // TODO: validate that this correct..never used endpoint besides their default before
-      apiKey: apiKey,
-    });
-    this.openaiClient = new OpenAIApi(configuration);
+
+    this.openaiClient = openaiClient;
   }
 
-  async createEmbedding(text: string) {
-    // TODO: do stuff with the openaiClient
-    return [1, 2, 3];
+  async createEmbedding({ text, userIp }: EmbeddingParams) {
+    const embeddingRequest = {
+      input: text,
+      user: userIp,
+    };
+    try {
+      const res = await this.openaiClient.embeddings.create(embeddingRequest);
+
+      const embeddings = res.data.data[0].embedding;
+      const { status } = res;
+
+      return { embeddings, status };
+    } catch (err: any) {
+      // Catch axios errors which occur if response 4XX or 5XX
+      if (err.response?.status && err.response?.data?.error) {
```

Comment:
Minor: this would be cleaner if you destructured response out of err so you don't need to keep repeating `err.response...` - I would even destructure status and data out of response if response is valid.

Reply:
destructured